Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp

```diff
@@ -2310,8 +2310,12 @@ PartialResult WARN_UNUSED_RETURN BBQJIT::addI64Add(Value lhs, Value rhs, Value&
             m_jit.add64(lhsLocation.asGPR(), rhsLocation.asGPR(), resultLocation.asGPR());
         ),
         BLOCK(
-            m_jit.move(ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
-            m_jit.add64(TrustedImm64(ImmHelpers::imm(lhs, rhs).asI64()), resultLocation.asGPR());
+            if (LIKELY(isRepresentableAs<int32_t>(ImmHelpers::imm(lhs, rhs).asI64())))
+                m_jit.add64(TrustedImm32(ImmHelpers::imm(lhs, rhs).asI64()), ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
+            else {
+                m_jit.move(ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
```

Comment:
We should add `void add64(TrustedImm64 imm, RegisterID src, RegisterID dest)` in MacroAssembler since this will generate `i64.add reg, -1` as `move reg, dst; add64 -1 dst`. I think we historically didn't need such a overload before Wasm as it's not possible to get an int64 in JS.

Reply:
`isRepresentableAs` works okay for signed integers, so you would truly have to exceed the 32-bit integer range to need the move; but point taken, next push contains the addition of 64-bit variants of these instructions in the macro assembler.